src/main/java/hudson/remoting/ProxyOutputStream.java

```diff
@@ -183,7 +183,6 @@ protected void finalize() throws Throwable {
         // if the object is auto-unexported, the export entry could have already been removed.
         if(channel!=null) {
```

Comment:
`&& oid != -1`

Reply:
yep though there is another issue. Not sure I want to touch it

Reply:
Well the null assignment was preventing a double export...

Reply:
double unexport is actually fine (excepting logging). But OK, will add a check